lib/galaxy/tools/parameters/basic.py

```diff
@@ -2457,7 +2395,7 @@ def to_dict( self, trans, view='collection', value_mapper=None, other_values=Non
         for hdca in self.match_multirun_collections( trans, history, dataset_matcher ):
             subcollection_type = self._history_query( trans ).can_map_over( hdca ).collection_type
             d['options']['hdca'].append({
-                'id': trans.security.encode_id( hdca.id ),
+                'id': trans.app.security.encode_id( hdca.id ),
```

Comment:
In general ``trans.security`` is superior to ``trans.app.security`` since it depends on the intereface of one [God object](https://en.wikipedia.org/wiki/God_object) instead of 2. Was it modified this way because the work context stuff doesn't have a reference to security? Seems like augmenting WorkContext to allow that would be a better approach or using ``self.app`` if that is available.